spec/ics-3-connection-semantics/README.md

```diff
@@ -0,0 +1,145 @@
+
+### Desired Properties
+
+- Implementing blockchains should be able to safely allow untrusted users to open and update connections.
+- The two connecting blockchains should be able to negotiate a shared connection "version" (agreeing on wire encoding format, accumulator proof format, etc.)
+
+#### Pre-Establishment
+
+Prior to connection establishment:
+
+- No further IBC subprotocols should operate, since cross-chain substates cannot be verified.
+- The initiating user (who creates the connection) must be able to specify a root-of-trust for the chain to connect to and a root of trust for the connecting chain (implicitly, e.g. by sending the transaction).
+
+#### During Handshake
+
+Once a negotiation handshake has begun:
+
+- Only the appropriate handshake datagrams can be executed in order.
+- No chain can masquerade as one of the handshaking chains (formalize...)
+
+#### Post-Establishment
+
+Once a negotiation handshake has completed:
+
+- The created connection objects on both chains contain the roots of trust specified by the initiating user.
+- No other connection objects can be maliciously created on other chains by replaying datagrams.
+- The connection should be able to be voluntarily & cleanly closed by both blockchains.
+- The connection should be able to be immediately closed upon discovery of a consensus equivocation.
+
+### Technical Specification
+
+#### Definitions
+
+This ICS defines the `Connection` type:
+
+![Datatypes](datatypes.png)
+
+Consensus-related primitives are as defined in [ICS 2: Consensus Requirements](../spec/ics-2-consensus-requirements).
+
+Accumulator-related primitives are as defined in [ICS 23: Cryptographic Accumulator](../spec/ics-23-cryptographic-accumulator).
+
+Versioning-related primitives are as defined in [ICS 6: Connection & Channel Versioning](../spec/ics-6-connection-channel-versioning).
+
+`Identifier` is an opaque value used as the key for a connection object; it must serialize to a bytestring. The identifier is not necessarily intended to be a human-readable name (and likely should not be, to discourage squatting or racing for identifiers). The opening handshake protocol allows each chain to verify the identifier used to reference the connection on the other chain, so the chains could choose to come to agreemnt on a common identifier. Further discussion is deferred to [ICS 10: Chain Naming Convention](../spec/ics-10-chain-naming-convention).
+
+#### Requirements
+
+Connection handlers and subsequent protocols make use of a simple key-value store interface provided by the underlying state machine. This store must provide three functions, which behave in the way you would expect:
```

Comment:
I don't think this is worth a separate ICS yet, but I could be convinced otherwise.